### ISSUE TITLE
IT test_can_bundle_configured_catalogs is slow

### ISSUE THREAD

**Issue by hannes-ucsc** [DataBiosphere/azul#5053]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/36301

The first test in that test case is fast:

```
test_can_bundle_canned_repository (integration_test.CanBundleScriptIntegrationTest) ... 2023-03-15 23:04:54,193 INFO    MainThread: Using random seed 8896398064767659320
2023-03-15 23:04:54,447 INFO    MainThread: Listing bundles with prefix '' in source CannedSourceRef(id='https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests', spec=SimpleSourceSpec(prefix=Prefix(common='', partition=0), name='https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests')).
2023-03-15 23:05:17,769 INFO    MainThread: There are 7 bundle(s) with prefix '' in source CannedSourceRef(id='https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests', spec=SimpleSourceSpec(prefix=Prefix(common='', partition=0), name='https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests')).
2023-03-15 23:05:17,770 INFO    MainThread: Canning bundle SourcedBundleFQID(uuid='4da04038-adab-59a9-b6c4-3a61242cc972', version='2021-01-01T00:00:00.000000Z', source=CannedSourceRef(id='https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests', spec=SimpleSourceSpec(prefix=Prefix(common='', partition=0), name='https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests'))) from catalog 'canned-it'
2023-03-15 23:05:17,780 INFO    MainThread: It took 0.001s to download bundle 4da04038-adab-59a9-b6c4-3a61242cc972.2021-01-01T00:00:00.000000Z
2023-03-15 23:05:17,780 INFO    MainThread: Fetched bundle '4da04038-adab-59a9-b6c4-3a61242cc972' version '2021-01-01T00:00:00.000000Z' from catalog 'canned-it'.
2023-03-15 23:05:17,781 INFO    MainThread: Successfully wrote /tmp/tmpspktslyu/4da04038-adab-59a9-b6c4-3a61242cc972.manifest.json
2023-03-15 23:05:17,782 INFO    MainThread: Successfully wrote /tmp/tmpspktslyu/4da04038-adab-59a9-b6c4-3a61242cc972.metadata.json
ok
```

The slow test starts

```
test_can_bundle_configured_catalogs (integration_test.CanBundleScriptIntegrationTest) ... 2023-03-15 23:05:17,783 INFO    MainThread: Using random seed 2295787861552327294
2023-03-15 23:05:17,800 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots?filter=hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12&limit=2, headers=None, timeout=20.0, body=None)
2023-03-15 23:05:18,772 DEBUG   MainThread: _request(…) -> 0.970s 200 b'{"total":697,"filteredTotal":1,"items":[{"id":"6d39533c-1f7f-494e-9cee-40ed2f801e19","name":"hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12","description":"Create snapshot hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2...'
2023-03-15 23:05:18,773 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots/6d39533c-1f7f-494e-9cee-40ed2f801e19, headers=None, timeout=20.0, body=None)
2023-03-15 23:05:20,866 DEBUG   MainThread: _request(…) -> 2.093s 200 b'{"id":"6d39533c-1f7f-494e-9cee-40ed2f801e19","name":"hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12","description":"Create snapshot hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12","createdDate":"2022-01-...'
2023-03-15 23:05:20,868 INFO    MainThread: Listing bundles with prefix '' in source SourceRef(id='6d39533c-1f7f-494e-9cee-40ed2f801e19', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-4f711011', name='hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12', is_snapshot=True)).
2023-03-15 23:05:20,869 DEBUG   MainThread: Query (205 characters total): "\n            SELECT links_id, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.links`\n            WHERE STARTS_WITH(links_id, '')\n        "
2023-03-15 23:05:25,082 DEBUG   MainThread: Job info: {"job_id": "c82c2c15-0cb1-4200-af9c-82dd39264051", "stats": {"estimatedBytesProcessed": "3864", "timeline": [{"elapsedMs": "1291", "totalSlotMs": "464", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3276", "totalSlotMs": "2569", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3469", "totalSlotMs": "2740", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "3864", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2740", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT links_id, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.links`\n            WHERE STARTS_WITH(links_id, '')\n        "}
2023-03-15 23:05:25,292 INFO    MainThread: There are 8 bundle(s) with prefix '' in source SourceRef(id='6d39533c-1f7f-494e-9cee-40ed2f801e19', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-4f711011', name='hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12', is_snapshot=True)).
2023-03-15 23:05:25,293 INFO    MainThread: Canning bundle SourcedBundleFQID(uuid='1b4b5aee-07bd-480f-9435-20d1c047b3c1', version='2021-06-21T20:04:30.265000Z', source=SourceRef(id='6d39533c-1f7f-494e-9cee-40ed2f801e19', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-4f711011', name='hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12', is_snapshot=True))) from catalog 'dcp24-it'
```

It picked a bundle so we must be in a sub test but, oddly, no `Beginning sub-test` message is logged.

And, for some reason it keeps getting snapshot information.

```
2023-03-15 23:05:25,333 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots?filter=hca_prod_027c51c60719469fa7f5640fe57cbece__20220110_dcp2_20220113_dcp12&limit=2, headers=None, timeout=20.0, body=None)
2023-03-15 23:05:25,805 DEBUG   MainThread: _request(…) -> 0.472s 200 b'{"total":697,"filteredTotal":1,"items":[{"id":"86273555-105f-4dde-8af1-bea78350b622","name":"hca_prod_027c51c60719469fa7f5640fe57cbece__20220110_dcp2_20220113_dcp12","description":"Create snapshot hca_prod_027c51c60719469fa7f5640fe57cbece__20220110_dcp2...'
```

This repeats until 5 min later …

```
2023-03-15 23:11:06,208 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots/f2be623d-b601-4d3b-a2e3-e45fa23bd8df, headers=None, timeout=20.0, body=None)
2023-03-15 23:11:08,239 DEBUG   MainThread: _request(…) -> 2.031s 200 b'{"id":"f2be623d-b601-4d3b-a2e3-e45fa23bd8df","name":"hca_prod_aff9c3cd6b844fc2abf2b9c0b3038277__20220330_dcp2_20220607_dcp17","description":"Create snapshot hca_prod_aff9c3cd6b844fc2abf2b9c0b3038277__20220330_dcp2_20220607_dcp17","createdDate":"2022-06-...'
```

… it finally starts retrieving the bundle it picked

```
2023-03-15 23:11:08,240 DEBUG   MainThread: Retrieving 1 entities of type 'links' ...
2023-03-15 23:11:08,240 DEBUG   MainThread: Query (408 characters total): '\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.links`\n            WHERE (links_id, version) IN ((\'1b4b5aee-07bd-480f-9435-20d1c047b3c1\', TIMESTAMP(\'2021-06-21T20:04:30.265000Z\')))\n        '
2023-03-15 23:11:11,541 DEBUG   MainThread: Job info: {"job_id": "4df22bc5-3bfc-4b66-a2b4-4cc7b2184512", "stats": {"estimatedBytesProcessed": "17288", "timeline": [{"elapsedMs": "1280", "totalSlotMs": "112", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2436", "totalSlotMs": "1737", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "1"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "17288", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "1737", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.links`\n            WHERE (links_id, version) IN (('1b4b5aee-07bd-480f-9435-20d1c047b3c1', TIMESTAMP('2021-06-21T20:04:30.265000Z')))\n        "}
2023-03-15 23:11:11,795 DEBUG   MainThread: Retrieved 1 entities of type 'links'
2023-03-15 23:11:11,796 INFO    MainThread: Bundle SourcedBundleFQID(uuid='1b4b5aee-07bd-480f-9435-20d1c047b3c1', version='2021-06-21T20:04:30.265000Z', source=SourceRef(id='6d39533c-1f7f-494e-9cee-40ed2f801e19', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-4f711011', name='hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12', is_snapshot=True))) is self-contained
2023-03-15 23:11:11,801 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'library_preparation_protocol' ...
2023-03-15 23:11:11,801 DEBUG   ThreadPoolExecutor-1_0: Query (414 characters total): '\n            SELECT library_preparation_protocol_id, JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.library_preparation_protocol`\n            WHERE (library_preparation_protocol_id) IN ((\'04907fd0-9f44-4098-8282-5fa22d4bb87c\'))\n        '
2023-03-15 23:11:15,940 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "fbad4a15-dbd5-490b-90da-fff2136669c9", "stats": {"estimatedBytesProcessed": "4544", "timeline": [{"elapsedMs": "1190", "totalSlotMs": "456", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "1974", "totalSlotMs": "1472", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3305", "totalSlotMs": "2933", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "0", "estimatedRunnableUnits": "1"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "4544", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2933", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT library_preparation_protocol_id, JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.library_preparation_protocol`\n            WHERE (library_preparation_protocol_id) IN (('04907fd0-9f44-4098-8282-5fa22d4bb87c'))\n        "}
2023-03-15 23:11:16,210 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'library_preparation_protocol'
2023-03-15 23:11:16,211 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'project' ...
2023-03-15 23:11:16,211 DEBUG   ThreadPoolExecutor-1_0: Query (351 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.project`\n            WHERE (project_id) IN ((\'d012d476-8f8c-4ff3-89d6-ebbe22c1b5c1\'))\n        '
2023-03-15 23:11:22,294 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "596e9a23-4b31-456b-9684-907f4838fbab", "stats": {"estimatedBytesProcessed": "6663", "timeline": [{"elapsedMs": "1255", "totalSlotMs": "115", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2243", "totalSlotMs": "1870", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "4244", "totalSlotMs": "3136", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "5334", "totalSlotMs": "3826", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "0", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "6663", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "3826", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.project`\n            WHERE (project_id) IN (('d012d476-8f8c-4ff3-89d6-ebbe22c1b5c1'))\n        "}
2023-03-15 23:11:22,508 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'project'
2023-03-15 23:11:22,509 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'donor_organism' ...
2023-03-15 23:11:22,509 DEBUG   ThreadPoolExecutor-1_0: Query (372 characters total): '\n            SELECT donor_organism_id, JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.donor_organism`\n            WHERE (donor_organism_id) IN ((\'02c2c861-55c5-4782-b2b2-685be75f6945\'))\n        '
2023-03-15 23:11:26,240 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "48067ba6-0042-45ef-a4a4-f720488470af", "stats": {"estimatedBytesProcessed": "4248", "timeline": [{"elapsedMs": "1286", "totalSlotMs": "422", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2941", "totalSlotMs": "1668", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "1"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "4248", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "1668", "cacheHit": false, "performanceInsights": {"avgPreviousExecutionMs": "190"}, "transferredBytes": "0"}, "query": "\n            SELECT donor_organism_id, JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.donor_organism`\n            WHERE (donor_organism_id) IN (('02c2c861-55c5-4782-b2b2-685be75f6945'))\n        "}
2023-03-15 23:11:26,489 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'donor_organism'
2023-03-15 23:11:26,489 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'analysis_protocol' ...
2023-03-15 23:11:26,489 DEBUG   ThreadPoolExecutor-1_0: Query (381 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, analysis_protocol_id, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.analysis_protocol`\n            WHERE (analysis_protocol_id) IN ((\'2aee97ee-fc3f-4b86-83a5-90a06899dc28\'))\n        '
2023-03-15 23:11:30,325 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "57e4b965-9231-43d2-8293-ff0ccfe3b5f2", "stats": {"estimatedBytesProcessed": "3969", "timeline": [{"elapsedMs": "1287", "totalSlotMs": "87", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2089", "totalSlotMs": "1117", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3096", "totalSlotMs": "1730", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "1"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "3969", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "1730", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, analysis_protocol_id, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.analysis_protocol`\n            WHERE (analysis_protocol_id) IN (('2aee97ee-fc3f-4b86-83a5-90a06899dc28'))\n        "}
2023-03-15 23:11:30,577 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'analysis_protocol'
2023-03-15 23:11:30,577 DEBUG   ThreadPoolExecutor-1_0: Retrieving 3 entities of type 'process' ...
2023-03-15 23:11:30,577 DEBUG   ThreadPoolExecutor-1_0: Query (435 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, process_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.process`\n            WHERE (process_id) IN ((\'d211f195-2be0-43b2-929c-e3f7775e04fa\'), (\'1b4b5aee-07bd-480f-9435-20d1c047b3c1\'), (\'ff2d9d2b-6d69-4069-b5ff-ff1d564cf516\'))\n        '
2023-03-15 23:11:34,656 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "5454180c-d49e-4d06-9a07-badef1552b54", "stats": {"estimatedBytesProcessed": "8445", "timeline": [{"elapsedMs": "1214", "totalSlotMs": "469", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2018", "totalSlotMs": "1783", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3422", "totalSlotMs": "2981", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "0", "estimatedRunnableUnits": "1"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "8445", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2981", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, process_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.process`\n            WHERE (process_id) IN (('d211f195-2be0-43b2-929c-e3f7775e04fa'), ('1b4b5aee-07bd-480f-9435-20d1c047b3c1'), ('ff2d9d2b-6d69-4069-b5ff-ff1d564cf516'))\n        "}
2023-03-15 23:11:34,897 DEBUG   ThreadPoolExecutor-1_0: Retrieved 3 entities of type 'process'
2023-03-15 23:11:34,897 DEBUG   ThreadPoolExecutor-1_0: Retrieving 2 entities of type 'dissociation_protocol' ...
2023-03-15 23:11:34,897 DEBUG   ThreadPoolExecutor-1_0: Query (435 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, dissociation_protocol_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.dissociation_protocol`\n            WHERE (dissociation_protocol_id) IN ((\'dddf356e-4f8a-4d8d-99a5-1341b30518ba\'), (\'4bfc00a2-3365-4544-b124-3db628a8e460\'))\n        '
2023-03-15 23:11:41,128 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "0ad926ce-b7c3-4aaf-9aea-3c2ddb94bbc8", "stats": {"estimatedBytesProcessed": "5347", "timeline": [{"elapsedMs": "1242", "totalSlotMs": "96", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2030", "totalSlotMs": "1268", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3231", "totalSlotMs": "2320", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "5231", "totalSlotMs": "4425", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "5400", "totalSlotMs": "4425", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "0", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "5347", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "4425", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, dissociation_protocol_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.dissociation_protocol`\n            WHERE (dissociation_protocol_id) IN (('dddf356e-4f8a-4d8d-99a5-1341b30518ba'), ('4bfc00a2-3365-4544-b124-3db628a8e460'))\n        "}
2023-03-15 23:11:41,359 DEBUG   ThreadPoolExecutor-1_0: Retrieved 2 entities of type 'dissociation_protocol'
2023-03-15 23:11:41,359 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'cell_suspension' ...
2023-03-15 23:11:41,359 DEBUG   ThreadPoolExecutor-1_0: Query (375 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, cell_suspension_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.cell_suspension`\n            WHERE (cell_suspension_id) IN ((\'4e5dab31-2cb7-4be3-bb52-ea0ef827ab62\'))\n        '
2023-03-15 23:11:45,229 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "218abe0d-4346-4a27-9d4e-2978eb379ea4", "stats": {"estimatedBytesProcessed": "4598", "timeline": [{"elapsedMs": "1116", "totalSlotMs": "313", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2923", "totalSlotMs": "3043", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3259", "totalSlotMs": "3662", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "4598", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "3662", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, cell_suspension_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.cell_suspension`\n            WHERE (cell_suspension_id) IN (('4e5dab31-2cb7-4be3-bb52-ea0ef827ab62'))\n        "}
2023-03-15 23:11:45,467 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'cell_suspension'
2023-03-15 23:11:45,467 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'specimen_from_organism' ...
2023-03-15 23:11:45,467 DEBUG   ThreadPoolExecutor-1_0: Query (396 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, specimen_from_organism_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.specimen_from_organism`\n            WHERE (specimen_from_organism_id) IN ((\'5ed3a59d-2b78-4152-8ad8-349c2c82b144\'))\n        '
2023-03-15 23:11:49,341 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "48bc965a-a368-4ee9-b4c5-5782f01e2f42", "stats": {"estimatedBytesProcessed": "5121", "timeline": [{"elapsedMs": "1250", "totalSlotMs": "365", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2057", "totalSlotMs": "1558", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3157", "totalSlotMs": "2580", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "5121", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2580", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, specimen_from_organism_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.specimen_from_organism`\n            WHERE (specimen_from_organism_id) IN (('5ed3a59d-2b78-4152-8ad8-349c2c82b144'))\n        "}
2023-03-15 23:11:49,615 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'specimen_from_organism'
2023-03-15 23:11:49,615 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'sequencing_protocol' ...
2023-03-15 23:11:49,615 DEBUG   ThreadPoolExecutor-1_0: Query (387 characters total): '\n            SELECT sequencing_protocol_id, JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.sequencing_protocol`\n            WHERE (sequencing_protocol_id) IN ((\'e1d77c81-fc47-4cde-8249-10d2593c05e3\'))\n        '
2023-03-15 23:11:53,015 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "1baac3c4-5ed3-473d-8d4b-b7f137785536", "stats": {"estimatedBytesProcessed": "4169", "timeline": [{"elapsedMs": "932", "totalSlotMs": "147", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2728", "totalSlotMs": "2542", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "4169", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2542", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT sequencing_protocol_id, JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.sequencing_protocol`\n            WHERE (sequencing_protocol_id) IN (('e1d77c81-fc47-4cde-8249-10d2593c05e3'))\n        "}
2023-03-15 23:11:53,244 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'sequencing_protocol'
2023-03-15 23:11:53,245 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'collection_protocol' ...
2023-03-15 23:11:53,245 DEBUG   ThreadPoolExecutor-1_0: Query (387 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version, collection_protocol_id\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.collection_protocol`\n            WHERE (collection_protocol_id) IN ((\'6a73cdcc-b094-490c-8180-f576f391cd4b\'))\n        '
2023-03-15 23:11:56,814 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "22a108bd-a9a3-4b8c-a92a-4b0b1b8b45f7", "stats": {"estimatedBytesProcessed": "3953", "timeline": [{"elapsedMs": "922", "totalSlotMs": "314", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "1681", "totalSlotMs": "1482", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2785", "totalSlotMs": "2341", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "3953", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2341", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, BYTE_LENGTH(content) AS content_size, version, collection_protocol_id\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.collection_protocol`\n            WHERE (collection_protocol_id) IN (('6a73cdcc-b094-490c-8180-f576f391cd4b'))\n        "}
2023-03-15 23:11:57,037 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'collection_protocol'
2023-03-15 23:11:57,037 DEBUG   ThreadPoolExecutor-1_0: Retrieving 1 entities of type 'analysis_file' ...
2023-03-15 23:11:57,037 DEBUG   ThreadPoolExecutor-1_0: Query (458 characters total): '\n            SELECT JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, file_id, descriptor, BYTE_LENGTH(content) AS content_size, analysis_file_id, version, JSON_EXTRACT_SCALAR(content, "$.file_core.file_name") AS file_name, content\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.analysis_file`\n            WHERE (analysis_file_id) IN ((\'21996ecc-931a-4775-80fb-f9b45f332bab\'))\n        '
2023-03-15 23:12:00,623 DEBUG   ThreadPoolExecutor-1_0: Job info: {"job_id": "4ab0d574-f357-4444-86b3-d6c6a1bb8a38", "stats": {"estimatedBytesProcessed": "17010", "timeline": [{"elapsedMs": "1160", "totalSlotMs": "352", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2881", "totalSlotMs": "2060", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "17010", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2060", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, file_id, descriptor, BYTE_LENGTH(content) AS content_size, analysis_file_id, version, JSON_EXTRACT_SCALAR(content, \"$.file_core.file_name\") AS file_name, content\n            FROM `datarepo-4f711011.hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12.analysis_file`\n            WHERE (analysis_file_id) IN (('21996ecc-931a-4775-80fb-f9b45f332bab'))\n        "}
2023-03-15 23:12:00,885 DEBUG   ThreadPoolExecutor-1_0: Retrieved 1 entities of type 'analysis_file'
2023-03-15 23:12:00,887 INFO    MainThread: It took 52.647s to download bundle 1b4b5aee-07bd-480f-9435-20d1c047b3c1.2021-06-21T20:04:30.265000Z
2023-03-15 23:12:00,887 INFO    MainThread: Fetched bundle '1b4b5aee-07bd-480f-9435-20d1c047b3c1' version '2021-06-21T20:04:30.265000Z' from catalog 'dcp24'.
2023-03-15 23:12:00,887 INFO    MainThread: Successfully wrote /tmp/tmpqivg9pwq/1b4b5aee-07bd-480f-9435-20d1c047b3c1.manifest.json
2023-03-15 23:12:00,889 INFO    MainThread: Successfully wrote /tmp/tmpqivg9pwq/1b4b5aee-07bd-480f-9435-20d1c047b3c1.metadata.json
```

This should be the end of one sub test and the beginning of the next. 

```
2023-03-15 23:12:00,893 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots?filter=hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1&limit=2, headers=None, timeout=20.0, body=None)
2023-03-15 23:12:01,363 DEBUG   MainThread: _request(…) -> 0.469s 200 b'{"total":697,"filteredTotal":1,"items":[{"id":"65f8b3ea-9aa5-4949-975a-69ec03ecebe7","name":"hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1","description":"Create snapshot hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_...'
2023-03-15 23:12:01,364 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots/65f8b3ea-9aa5-4949-975a-69ec03ecebe7, headers=None, timeout=20.0, body=None)
2023-03-15 23:12:03,418 DEBUG   MainThread: _request(…) -> 2.053s 200 b'{"id":"65f8b3ea-9aa5-4949-975a-69ec03ecebe7","name":"hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1","description":"Create snapshot hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1","createdDate":"2022-01-04...'
2023-03-15 23:12:03,419 INFO    MainThread: Listing bundles with prefix 'f' in source SourceRef(id='65f8b3ea-9aa5-4949-975a-69ec03ecebe7', spec=TDRSourceSpec(prefix=Prefix(common='', partition=1), project='datarepo-0e566a81', name='hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1', is_snapshot=True)).
2023-03-15 23:12:03,419 DEBUG   MainThread: Query (205 characters total): "\n            SELECT links_id, version\n            FROM `datarepo-0e566a81.hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1.links`\n            WHERE STARTS_WITH(links_id, 'f')\n        "
2023-03-15 23:12:07,366 DEBUG   MainThread: Job info: {"job_id": "b79277ed-f87e-44ac-b9a4-b8a1fb938b6e", "stats": {"estimatedBytesProcessed": "1633080", "timeline": [{"elapsedMs": "925", "totalSlotMs": "448", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2613", "totalSlotMs": "2711", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3124", "totalSlotMs": "3606", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "1633080", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "3606", "cacheHit": false, "performanceInsights": {"avgPreviousExecutionMs": "310"}, "transferredBytes": "0"}, "query": "\n            SELECT links_id, version\n            FROM `datarepo-0e566a81.hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1.links`\n            WHERE STARTS_WITH(links_id, 'f')\n        "}
2023-03-15 23:12:07,714 INFO    MainThread: There are 205 bundle(s) with prefix 'f' in source SourceRef(id='65f8b3ea-9aa5-4949-975a-69ec03ecebe7', spec=TDRSourceSpec(prefix=Prefix(common='', partition=1), project='datarepo-0e566a81', name='hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1', is_snapshot=True)).
2023-03-15 23:12:07,714 INFO    MainThread: Canning bundle SourcedBundleFQID(uuid='f273118d-d72a-4f7e-bdf3-fd1e06f711fc', version='2019-05-14T10:50:20.477000Z', source=SourceRef(id='65f8b3ea-9aa5-4949-975a-69ec03ecebe7', spec=TDRSourceSpec(prefix=Prefix(common='', partition=1), project='datarepo-0e566a81', name='hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1', is_snapshot=True))) from catalog 'dcp1-it'
```

It picked another bundle but keeps getting snapshot info …

```
2023-03-15 23:12:07,737 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots?filter=hca_prod_c6a50b2a3dfd4ca89b483e682f568a25__20220303_dcp2_20220307_dcp14&limit=2, headers=None, timeout=20.0, body=None)
2023-03-15 23:12:08,134 DEBUG   MainThread: _request(…) -> 0.397s 200 b'{"total":697,"filteredTotal":1,"items":[{"id":"d3388230-361d-420c-9483-4832c7beb64e","name":"hca_prod_c6a50b2a3dfd4ca89b483e682f568a25__20220303_dcp2_20220307_dcp14","description":"Create snapshot hca_prod_c6a50b2a3dfd4ca89b483e682f568a25__20220303_dcp2...'

…

2023-03-15 23:20:08,713 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots/9bee685e-78d4-4ba0-9804-d811806bf7c0, headers=None, timeout=20.0, body=None)
2023-03-15 23:20:10,750 DEBUG   MainThread: _request(…) -> 2.037s 200 b'{"id":"9bee685e-78d4-4ba0-9804-d811806bf7c0","name":"hca_prod_4d6f6c962a8343d88fe10f53bffd4674__20211129_dcp1_20211129_dcp1","description":"Create snapshot hca_prod_4d6f6c962a8343d88fe10f53bffd4674__20211129_dcp1_20211129_dcp1","createdDate":"2021-11-29...'
```

… this time, for eight minutes. It fetches the bundle …

```
2023-03-15 23:20:10,751 DEBUG   MainThread: Retrieving 1 entities of type 'links' ...
2023-03-15 23:20:10,752 DEBUG   MainThread: Query (407 characters total): '\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-0e566a81.hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1.links`\n            WHERE (links_id, version) IN ((\'f273118d-d72a-4f7e-bdf3-fd1e06f711fc\', TIMESTAMP(\'2019-05-14T10:50:20.477000Z\')))\n        '
2023-03-15 23:20:14,949 DEBUG   MainThread: Job info: {"job_id": "213c2c6d-6ece-42a9-8de1-f7585f9caf88", "stats": {"estimatedBytesProcessed": "6559708", "timeline": [{"elapsedMs": "1320", "totalSlotMs": "469", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2270", "totalSlotMs": "1521", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3365", "totalSlotMs": "2568", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "6559708", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2568", "cacheHit": false, "performanceInsights": {"avgPreviousExecutionMs": "223"}, "transferredBytes": "0"}, "query": "\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-0e566a81.hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1.links`\n            WHERE (links_id, version) IN (('f273118d-d72a-4f7e-bdf3-fd1e06f711fc', TIMESTAMP('2019-05-14T10:50:20.477000Z')))\n        "}
2023-03-15 23:20:15,205 DEBUG   MainThread: Retrieved 1 entities of type 'links'
2023-03-15 23:20:15,206 INFO    MainThread: Bundle SourcedBundleFQID(uuid='f273118d-d72a-4f7e-bdf3-fd1e06f711fc', version='2019-05-14T10:50:20.477000Z', source=SourceRef(id='65f8b3ea-9aa5-4949-975a-69ec03ecebe7', spec=TDRSourceSpec(prefix=Prefix(common='', partition=1), project='datarepo-0e566a81', name='hca_prod_ae71be1dddd84feb9bed24c3ddb6e1ad__20211213_dcp1_20220104_dcp1', is_snapshot=True))) is self-contained
2023-03-15 23:20:15,208 DEBUG   ThreadPoolExecutor-2_0: Retrieving 3 entities of type 'process' ...

…

2023-03-15 23:20:44,155 DEBUG   ThreadPoolExecutor-2_0: Retrieved 1 entities of type 'dissociation_protocol'
2023-03-15 23:20:44,157 INFO    MainThread: It took 33.406s to download bundle f273118d-d72a-4f7e-bdf3-fd1e06f711fc.2019-05-14T10:50:20.477000Z
2023-03-15 23:20:44,158 INFO    MainThread: Fetched bundle 'f273118d-d72a-4f7e-bdf3-fd1e06f711fc' version '2019-05-14T10:50:20.477000Z' from catalog 'dcp1'.
2023-03-15 23:20:44,159 INFO    MainThread: Successfully wrote /tmp/tmpi_19nl0t/f273118d-d72a-4f7e-bdf3-fd1e06f711fc.manifest.json
2023-03-15 23:20:44,161 INFO    MainThread: Successfully wrote /tmp/tmpi_19nl0t/f273118d-d72a-4f7e-bdf3-fd1e06f711fc.metadata.json
```

This repeats (though not as extreme)

```
2023-03-15 23:20:44,165 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots?filter=lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2&limit=2, headers=None, timeout=20.0, body=None)
2023-03-15 23:20:44,976 DEBUG   MainThread: _request(…) -> 0.811s 200 b'{"total":697,"filteredTotal":1,"items":[{"id":"4e6b4bd3-8534-4ebf-868b-8a3ed7394f24","name":"lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2","description":"Create snapshot lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_202...'
2023-03-15 23:20:44,977 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots/4e6b4bd3-8534-4ebf-868b-8a3ed7394f24, headers=None, timeout=20.0, body=None)
2023-03-15 23:20:47,028 DEBUG   MainThread: _request(…) -> 2.050s 200 b'{"id":"4e6b4bd3-8534-4ebf-868b-8a3ed7394f24","name":"lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2","description":"Create snapshot lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2","createdDate":"2022-03-14T18:...'
2023-03-15 23:20:47,029 INFO    MainThread: Listing bundles with prefix '' in source SourceRef(id='4e6b4bd3-8534-4ebf-868b-8a3ed7394f24', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-5eee9956', name='lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2', is_snapshot=True)).
2023-03-15 23:20:47,029 DEBUG   MainThread: Query (202 characters total): "\n            SELECT links_id, version\n            FROM `datarepo-5eee9956.lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2.links`\n            WHERE STARTS_WITH(links_id, '')\n        "
2023-03-15 23:20:51,052 DEBUG   MainThread: Job info: {"job_id": "8f02fddf-ff70-47aa-9d52-0be7770b6f0c", "stats": {"estimatedBytesProcessed": "19092", "timeline": [{"elapsedMs": "1152", "totalSlotMs": "336", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2143", "totalSlotMs": "2031", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3360", "totalSlotMs": "2830", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "1"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "19092", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2830", "cacheHit": false, "performanceInsights": {"avgPreviousExecutionMs": "260"}, "transferredBytes": "0"}, "query": "\n            SELECT links_id, version\n            FROM `datarepo-5eee9956.lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2.links`\n            WHERE STARTS_WITH(links_id, '')\n        "}
2023-03-15 23:20:51,266 INFO    MainThread: There are 1 bundle(s) with prefix '' in source SourceRef(id='4e6b4bd3-8534-4ebf-868b-8a3ed7394f24', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-5eee9956', name='lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2', is_snapshot=True)).
2023-03-15 23:20:51,267 INFO    MainThread: Canning bundle SourcedBundleFQID(uuid='a078a6cb-a72a-305c-80df-cf35aedd01ff', version='2021-09-27T16:41:49.288213Z', source=SourceRef(id='4e6b4bd3-8534-4ebf-868b-8a3ed7394f24', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-5eee9956', name='lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2', is_snapshot=True))) from catalog 'lm2-it'
2023-03-15 23:20:51,322 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots?filter=hca_prod_2043c65a1cf84828a6569e247d4e64f1__20211129_dcp1_20220126_dcp1&limit=2, headers=None, timeout=20.0, body=None)
2023-03-15 23:20:51,791 DEBUG   MainThread: _request(…) -> 0.469s 200 b'{"total":697,"filteredTotal":1,"items":[{"id":"0cd5cc9a-a433-4d65-b94c-32b049b23d7e","name":"hca_prod_2043c65a1cf84828a6569e247d4e64f1__20211129_dcp1_20220126_dcp1","description":"Create snapshot hca_prod_2043c65a1cf84828a6569e247d4e64f1__20211129_dcp1_...'

…

2023-03-15 23:21:41,589 DEBUG   MainThread: _request('GET', https://data.terra.bio/api/repository/v1/snapshots/07c188d1-fcff-4ffa-b0f1-3374ce824b8a, headers=None, timeout=20.0, body=None)
2023-03-15 23:21:43,768 DEBUG   MainThread: _request(…) -> 2.179s 200 b'{"id":"07c188d1-fcff-4ffa-b0f1-3374ce824b8a","name":"lungmap_prod_1bdcecde16be420888f478cd2133d11d__20220308_20220308","description":"Create snapshot lungmap_prod_1bdcecde16be420888f478cd2133d11d__20220308_20220308","createdDate":"2022-03-08T20:26:14.16...'
2023-03-15 23:21:43,769 DEBUG   MainThread: Retrieving 1 entities of type 'links' ...
2023-03-15 23:21:43,769 DEBUG   MainThread: Query (405 characters total): '\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-5eee9956.lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2.links`\n            WHERE (links_id, version) IN ((\'a078a6cb-a72a-305c-80df-cf35aedd01ff\', TIMESTAMP(\'2021-09-27T16:41:49.288213Z\')))\n        '
2023-03-15 23:21:46,798 DEBUG   MainThread: Job info: {"job_id": "89e56063-110d-4860-8bdb-7b2f13f3cd18", "stats": {"estimatedBytesProcessed": "119052", "timeline": [{"elapsedMs": "825", "totalSlotMs": "974", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2315", "totalSlotMs": "2037", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "119052", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2037", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-5eee9956.lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2.links`\n            WHERE (links_id, version) IN (('a078a6cb-a72a-305c-80df-cf35aedd01ff', TIMESTAMP('2021-09-27T16:41:49.288213Z')))\n        "}
2023-03-15 23:21:47,036 DEBUG   MainThread: Retrieved 1 entities of type 'links'
2023-03-15 23:21:47,047 INFO    MainThread: Bundle SourcedBundleFQID(uuid='a078a6cb-a72a-305c-80df-cf35aedd01ff', version='2021-09-27T16:41:49.288213Z', source=SourceRef(id='4e6b4bd3-8534-4ebf-868b-8a3ed7394f24', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-5eee9956', name='lungmap_prod_00f056f273ff43ac97ff69ca10e38c89__20220308_20220314_lm2', is_snapshot=True))) is self-contained
2023-03-15 23:21:47,048 DEBUG   ThreadPoolExecutor-3_0: Retrieving 11 entities of type 'analysis_file' ...

…

2023-03-15 23:22:35,591 DEBUG   ThreadPoolExecutor-3_0: Retrieved 1 entities of type 'dissociation_protocol'
2023-03-15 23:22:35,592 INFO    MainThread: It took 51.823s to download bundle a078a6cb-a72a-305c-80df-cf35aedd01ff.2021-09-27T16:41:49.288213Z
2023-03-15 23:22:35,592 INFO    MainThread: Fetched bundle 'a078a6cb-a72a-305c-80df-cf35aedd01ff' version '2021-09-27T16:41:49.288213Z' from catalog 'lm2'.
2023-03-15 23:22:35,602 INFO    MainThread: Successfully wrote /tmp/tmpghypac7w/a078a6cb-a72a-305c-80df-cf35aedd01ff.manifest.json
2023-03-15 23:22:35,626 INFO    MainThread: Successfully wrote /tmp/tmpghypac7w/a078a6cb-a72a-305c-80df-cf35aedd01ff.metadata.json
```

and again

```
2023-03-15 23:22:35,636 INFO    MainThread: Listing bundles with prefix '' in source SourceRef(id='87f88695-79d5-40ba-8a88-0a97aaf2b2ad', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-cfaedae8', name='lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308', is_snapshot=True)).
2023-03-15 23:22:35,636 DEBUG   MainThread: Query (198 characters total): "\n            SELECT links_id, version\n            FROM `datarepo-cfaedae8.lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308.links`\n            WHERE STARTS_WITH(links_id, '')\n        "
2023-03-15 23:22:39,727 DEBUG   MainThread: Job info: {"job_id": "808699fa-c88f-4de6-b767-914bdc7e78e6", "stats": {"estimatedBytesProcessed": "15968", "timeline": [{"elapsedMs": "931", "totalSlotMs": "154", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "1918", "totalSlotMs": "1502", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "3165", "totalSlotMs": "2605", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "0"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "15968", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2605", "cacheHit": false, "performanceInsights": {"avgPreviousExecutionMs": "272", "stagePerformanceChangeInsights": [{"stageId": "2", "inputDataChange": {"recordsReadDiffPercentage": 100}}]}, "transferredBytes": "0"}, "query": "\n            SELECT links_id, version\n            FROM `datarepo-cfaedae8.lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308.links`\n            WHERE STARTS_WITH(links_id, '')\n        "}
2023-03-15 23:22:39,978 INFO    MainThread: There are 1 bundle(s) with prefix '' in source SourceRef(id='87f88695-79d5-40ba-8a88-0a97aaf2b2ad', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-cfaedae8', name='lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308', is_snapshot=True)).
2023-03-15 23:22:39,978 INFO    MainThread: Canning bundle SourcedBundleFQID(uuid='153f5c7a-593e-37fc-aa56-4870057f27c7', version='2021-09-17T17:12:19.996404Z', source=SourceRef(id='87f88695-79d5-40ba-8a88-0a97aaf2b2ad', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-cfaedae8', name='lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308', is_snapshot=True))) from catalog 'lm3-it'
2023-03-15 23:22:40,044 DEBUG   MainThread: Retrieving 1 entities of type 'links' ...
2023-03-15 23:22:40,044 DEBUG   MainThread: Query (401 characters total): '\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, "$.schema_type") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-cfaedae8.lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308.links`\n            WHERE (links_id, version) IN ((\'153f5c7a-593e-37fc-aa56-4870057f27c7\', TIMESTAMP(\'2021-09-17T17:12:19.996404Z\')))\n        '
2023-03-15 23:22:43,607 DEBUG   MainThread: Job info: {"job_id": "bc1b901e-2ee4-440c-a7b5-3c81e2e9ab6b", "stats": {"estimatedBytesProcessed": "57729", "timeline": [{"elapsedMs": "919", "totalSlotMs": "213", "pendingUnits": "2", "completedUnits": "0", "activeUnits": "2", "estimatedRunnableUnits": "0"}, {"elapsedMs": "1688", "totalSlotMs": "1662", "pendingUnits": "1", "completedUnits": "1", "activeUnits": "1", "estimatedRunnableUnits": "0"}, {"elapsedMs": "2699", "totalSlotMs": "2581", "pendingUnits": "0", "completedUnits": "2", "activeUnits": "1", "estimatedRunnableUnits": "1"}], "totalPartitionsProcessed": "1", "totalBytesProcessed": "57729", "totalBytesBilled": "20971520", "billingTier": 1, "totalSlotMs": "2581", "cacheHit": false, "transferredBytes": "0"}, "query": "\n            SELECT links_id, JSON_EXTRACT_SCALAR(content, \"$.schema_type\") AS schema_type, content, project_id, BYTE_LENGTH(content) AS content_size, version\n            FROM `datarepo-cfaedae8.lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308.links`\n            WHERE (links_id, version) IN (('153f5c7a-593e-37fc-aa56-4870057f27c7', TIMESTAMP('2021-09-17T17:12:19.996404Z')))\n        "}
2023-03-15 23:22:43,814 DEBUG   MainThread: Retrieved 1 entities of type 'links'
2023-03-15 23:22:43,817 INFO    MainThread: Bundle SourcedBundleFQID(uuid='153f5c7a-593e-37fc-aa56-4870057f27c7', version='2021-09-17T17:12:19.996404Z', source=SourceRef(id='87f88695-79d5-40ba-8a88-0a97aaf2b2ad', spec=TDRSourceSpec(prefix=Prefix(common='', partition=0), project='datarepo-cfaedae8', name='lungmap_prod_2620497955a349b28d2b53e0bdfcb176__20220308_20220308', is_snapshot=True))) is self-contained
2023-03-15 23:22:43,818 DEBUG   ThreadPoolExecutor-4_0: Retrieving 69 entities of type 'process' ...

…

2023-03-15 23:23:28,556 DEBUG   ThreadPoolExecutor-4_0: Retrieved 1 entities of type 'collection_protocol'
2023-03-15 23:23:28,558 INFO    MainThread: It took 48.514s to download bundle 153f5c7a-593e-37fc-aa56-4870057f27c7.2021-09-17T17:12:19.996404Z
2023-03-15 23:23:28,558 INFO    MainThread: Fetched bundle '153f5c7a-593e-37fc-aa56-4870057f27c7' version '2021-09-17T17:12:19.996404Z' from catalog 'lm2'.
2023-03-15 23:23:28,568 INFO    MainThread: Successfully wrote /tmp/tmp8n019kus/153f5c7a-593e-37fc-aa56-4870057f27c7.manifest.json
2023-03-15 23:23:28,585 INFO    MainThread: Successfully wrote /tmp/tmp8n019kus/153f5c7a-593e-37fc-aa56-4870057f27c7.metadata.json
ok
```

Nineteen minutes in we finally get to the next test. But the job eventually times out:

```
ERROR: Job failed: execution took longer than 1h0m0s seconds
```

It shouldn't take that long to pick and can a bundle for each IT catalog.

It's also puzzling as to why the sub-test boundaries aren't logged. But that may be another issue.

**Comment by nadove-ucsc** [comment]
I have a hypothesis, but if I'm correct, it will be very difficult to prove.

https://github.com/DataBiosphere/azul/blob/317e856621e1670149b7d524a41181b0a309b155/scripts/can_bundle.py#L79

When fetching a bundle, the `can_bundle` script iterates over every source in the specified catalog, calling `resolve_source` on every iteration, until it finds a source that matches the specified source. The order in which the sources are processed is non-deterministic (iteration is performed on a set). There are no log statements within this method. My hypothesis is that we got unlucky with the correct source not occurring until near the end of the loop for each catalog.

I don't think these calls to `resolve_source` are necessary. The purpose of `resolve_source` is to obtain instances of `SourceRef`, but instances of `SourceSpec` are sufficient for determining when to exit the loop:

https://github.com/DataBiosphere/azul/blob/317e856621e1670149b7d524a41181b0a309b155/scripts/can_bundle.py#L90

The plugin is already configured with instances of `SourceSpec`, so we should be able to remove all network requests from this loop, and just call `resolve_source` twice*, once before and once after the loop is finished. 

I don't have any idea why the `subTest` contexts are not logged.

*I'm not convinced the pre-loop call is actually necessary either.

**Comment by hannes-ucsc** [comment]
For demo, show that test is significantly faster in `prod`.

**Comment by dsotirho-ucsc** [comment]
Reoccurrence in `prod` IT.
`test_can_bundle_configured_catalogs` took 19 minutes.
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/36709

```
2023-03-23 21:26:39,229 : IT started
…
2023-03-23 21:26:52,122 : test_can_bundle_canned_repository starts
2023-03-23 21:27:20,341 : test_can_bundle_configured_catalogs starts
2023-03-23 21:46:07,832 : test_catalog_listing starts
…
2023-03-23 22:26:13,557 : Job failed: execution took longer than 1h0m0s seconds
```

**Comment by dsotirho-ucsc** [comment]
Reoccurrence on `prod` IT
`test_can_bundle_configured_catalogs` took 27 minutes
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/36777
